sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.ts

```diff
@@ -432,4 +422,44 @@ export class StudentAESTController extends BaseController {
       userToken.userId,
     );
   }
+
+  /**
+   * Updates the student info (lastname, givenNames, dob, email)
+   * for a basic BCeID student.
+   * @param studentId related student id.
+   * @param payload payload to be updated.
+   */
+  @Roles(Role.StudentEditProfile)
+  @Patch(":studentId")
+  @ApiNotFoundResponse({ description: "Student does not exist." })
+  @ApiUnprocessableEntityResponse({
+    description: "No profile update found to be saved.",
+  })
+  async updateProfileInformation(
+    @Param("studentId", ParseIntPipe) studentId: number,
+    @Body() payload: UpdateStudentDetailsAPIInDTO,
+    @UserToken() userToken: IUserToken,
+  ): Promise<void> {
+    const student = await this.studentService.getStudentById(studentId);
+    if (!student) {
+      throw new NotFoundException("Student does not exist.");
+    }
+    if (student.user.identityProviderType !== IdentityProviders.BCeIDBasic) {
+      throw new UnprocessableEntityException(
+        "Not possible to update a non-basic-BCeID student.",
+      );
+    }
+    const updated = await this.studentService.updateStudentUserData(
+      {
+        studentId,
+        ...payload,
+      },
+      userToken.userId,
+    );
+    if (!updated) {
+      throw new UnprocessableEntityException(
+        "No profile update found to be saved.",
```

Comment:
The message is misleading, the `!updated` means "No profile data updated because no changes were detected.". Can you please update it?

Reply:
The message "No profile data updated because no changes were detected." should be updated in the `@ApiUnprocessableEntityResponse`.